--- a/packages/apps/Updater/app/src/main/java/org/evolution/updater/controller/UpdaterController.java
+++ b/packages/apps/Updater/app/src/main/java/org/evolution/updater/controller/UpdaterController.java
@@ -338,7 +338,15 @@
             if (entry != null) {
                 Update updateAdded = entry.mUpdate;
                 updateAdded.setAvailableOnline(availableOnline && updateAdded.getAvailableOnline());
+                updateAdded.setName(updateInfo.getName());
                 updateAdded.setDownloadUrl(updateInfo.getDownloadUrl());
+                updateAdded.setTimestamp(updateInfo.getTimestamp());
+                updateAdded.setType(updateInfo.getType());
+                updateAdded.setVersion(updateInfo.getVersion());
+                updateAdded.setFileSize(updateInfo.getFileSize());
+                updateAdded.setWebSource(updateInfo.getWebSource());
+                updateAdded.setDescription(updateInfo.getDescription());
+                updateAdded.setChangelogUrl(updateInfo.getChangelogUrl());
             }
             return false;
         }
