--- a/frameworks/base/packages/SystemUI/src/com/android/systemui/keyguard/ui/view/layout/sections/AodBurnInLayer.kt
+++ b/frameworks/base/packages/SystemUI/src/com/android/systemui/keyguard/ui/view/layout/sections/AodBurnInLayer.kt
@@ -21,6 +21,7 @@
 import android.view.ViewTreeObserver.OnPreDrawListener
 import androidx.constraintlayout.helper.widget.Layer
 import androidx.constraintlayout.widget.ConstraintLayout
+import java.lang.reflect.Field
 
 class AodBurnInLayer(
     context: Context,
@@ -44,6 +45,7 @@
         super.setScaleY(_scaleY)
         super.setTranslationX(_translationX)
         super.setTranslationY(_translationY)
+        invalidateViewsCache()
         super.setRotation(_rotation)
         true
     }
@@ -76,6 +78,15 @@
         (parent as? ConstraintLayout)?.let { super.updatePostLayout(it) }
     }
 
+    private fun invalidateViewsCache() {
+        try {
+            val field: Field = Layer::class.java.getDeclaredField("mViews")
+            field.isAccessible = true
+            field.set(this, null)
+        } catch (_: Exception) {
+        }
+    }
+
     override fun setScaleX(scaleX: Float) {
         _scaleX = scaleX
         super.setScaleX(scaleX)
