--- a/frameworks/base/services/java/com/android/server/SystemServer.java
+++ b/frameworks/base/services/java/com/android/server/SystemServer.java
@@ -1821,6 +1821,10 @@
             mSystemServiceManager.startService(EvolutionSystemExService.class);
             t.traceEnd();
 
+            t.traceBegin("StartDtwSystemService");
+            mSystemServiceManager.startService(DtwSystemService.class);
+            t.traceEnd();
+
             // Start receiving calls from SensorManager services. Start in a separate thread
             // because it need to connect to SensorManager. This has to start
             // after PHASE_WAIT_FOR_SENSOR_SERVICE is done.
